internal/bundlereader/helm_test.go

```diff
@@ -0,0 +1,302 @@
+package bundlereader_test
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/subtle"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/rancher/fleet/internal/bundlereader"
+	"github.com/rancher/fleet/internal/mocks"
+	fleet "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+)
+
+const (
+	authUsername  = "holadonpepito"
+	authPassword  = "holadonjose"
+	helmRepoIndex = `apiVersion: v1
+entries:
+  sleeper:
+    - created: 2016-10-06T16:23:20.499814565-06:00
+      description: Super sleeper chart
+      digest: 99c76e403d752c84ead610644d4b1c2f2b453a74b921f422b9dcb8a7c8b559cd
+      home: https://helm.sh/helm
+      name: alpine
+      sources:
+      - https://github.com/helm/helm
+      urls:
+      - https://##URL##/sleeper-chart-0.1.0.tgz
+      version: 0.1.0
+generated: 2016-10-06T16:23:20.499029981-06:00`
+)
+
+func checksumPrefix(helm *fleet.HelmOptions) string {
+	if helm == nil {
+		return "none"
+	}
+	return fmt.Sprintf(".chart/%x", sha256.Sum256([]byte(helm.Chart + ":" + helm.Repo + ":" + helm.Version)[:]))
+}
```

Comment:
I did not want to export a so simple function just because we need it for testing. 

I didn't see a reason for exporting that function aside from that (I see that function as some internal aspect of the package) and... if the function changes in the future the test will fail and it's pretty straightforward to change.